python/ycm/vimsupport.py

```diff
@@ -503,3 +518,29 @@ def ReplaceChunk( start, end, replacement_text, line_delta, char_delta,
 
   new_line_delta = replacement_lines_count - source_lines_count
   return ( new_line_delta, new_char_delta )
+
+
+def InsertNamespace( namespace ):
+  if VariableExists( 'g:ycm_csharp_insert_namespace_expr' ):
```

Comment:
This function seems csharp-specific. Perhaps more of the cs-specific stuff should be in `OnCompleteDone_Csharp` ? Or maybe just call the method `InsertCsharpNamespace` ? Reason is the eclim does this sort of thing and I suppose if we ever have a java completer, it would need to do similar (but not identical) processing.